orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/MonitoredDeployStrategy.groovy

```diff
@@ -0,0 +1,354 @@
+    if (deploySteps.isEmpty() || deploySteps[-1] != 100) {
+      deploySteps.add(100)
+    }
+
+    // Get source ASG from prior determineSourceServerGroupTask
+    def source = null
+
+    try {
+      StageData.Source sourceServerGroup
+
+      Stage parentCreateServerGroupStage = stage.directAncestors().find() { it.type == CreateServerGroupStage.PIPELINE_CONFIG_TYPE }
+
+      if (parentCreateServerGroupStage.status == ExecutionStatus.NOT_STARTED) {
+        // No point in composing the flow if we are called to plan "beforeStages" since we don't have any STAGE_BEFOREs.
+        // Also, we rely on the the source server group task to have run already.
+        // In the near future we will move composeFlow into beforeStages and afterStages instead of the
+        // deprecated aroundStages
+        return []
+      }
+
+      StageData parentStageData = parentCreateServerGroupStage.mapTo(StageData)
+      sourceServerGroup = parentStageData.source
+
+      if (sourceServerGroup != null && sourceServerGroup.serverGroupName != null) {
+        source = new ResizeStrategy.Source(
+          region: sourceServerGroup.region,
+          serverGroupName: sourceServerGroup.serverGroupName,
+          credentials: stageData.credentials ?: stageData.account,
+          cloudProvider: stageData.cloudProvider
+        )
+      }
+    } catch (Exception e) {
+      // This probably means there was no parent CreateServerGroup stage - which should never happen
+      throw new IllegalStateException("Failed to determine source server group from parent stage while planning Monitored Deploy flow", e)
+    }
+
+    MonitoredDeployStageData mdsd = stage.mapTo(MonitoredDeployStageData.class)
+
+    // TODO(mvulfson): I don't love this
+    def evalContext = mdsd.getChildStageContext()
+    MonitoredDeployInternalStageData internalStageData = new MonitoredDeployInternalStageData()
+    internalStageData.account = baseContext.credentials
+    internalStageData.cloudProvider = baseContext.cloudProvider
+    internalStageData.region = baseContext.region
+    internalStageData.oldServerGroup = source?.serverGroupName
+    internalStageData.newServerGroup = mdsd.deployServerGroups[baseContext.region].first()
+    internalStageData.parameters = mdsd.deploymentMonitor.parameters
+
+    evalContext += internalStageData.toContextMap()
+
+    def findContext = baseContext + [
+      //target        : TargetServerGroup.Params.Target.current_asg_dynamic,
+      serverGroupName: internalStageData.newServerGroup,
+      targetLocation: cleanupConfig.location,
+    ]
+
+    stages << newStage(
+      stage.execution,
+      determineTargetServerGroupStage.type,
+      "Determine Deployed Server Group",
+      findContext,
+      stage,
+      SyntheticStageOwner.STAGE_AFTER
+    )
+
+    if (source == null) {
+      log.warn("no source server group -- will perform Monitored Deploy to exact fallback capacity $savedCapacity with no disableCluster or scaleDownCluster stages")
```

Comment:
Nice! Maybe `info` more than `warn` since it's not a problem?

Reply:
copied from RRB, but yeah, `info` is better, thanks - note this file will be "cleaned up a lot" for now it's mostly a copy of the rrb.groovy :(